### PR TITLE
docs(all): add toc to README.md and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ijemmy**
## Description of your changes

Add table of contents to CONTRIBUTING.md and README.md

Both files are large and hard to navigate. Adding ToC will help readers to see the over all structure and can navigate to the relevant content quicker.

### How to verify this change
Preview the Markdown File and click on the links in ToC.

### Related issues, RFCs
N/A

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [ ] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [ ] I have *added tests* that prove my change is effective and works
- [ ] New and existing *unit tests pass* locally and in Github Actions
- [ ] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dreamorosi — APPROVED** [review]
Thanks for the PR!

**Comment by orozcoadrian** [comment]
Reminder that github has a built-in (and automatically-generated/always up-to-date) ToC ux feature.

Having multiple ways to navigate is probably ok, we'll just need to remember to check if the ToC needs updating whenever content in the page changes.

image

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,6 +6,39 @@ documentation, we greatly value feedback and contributions from our community.
 Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
 information to effectively respond to your bug report or contribution.
 
+## Table of contents
+
+- [Contributing Guidelines](#contributing-guidelines)
+  - [Table of contents](#table-of-contents)
```
Comment (saragerion) [inline CONTRIBUTING.md:12]:
```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,6 +6,39 @@ documentation, we greatly value feedback and contributions from our community.
...
+    - [Step 3: Work your Magic](#step-3-work-your-magic)
+    - [Step 4: Pull Request](#step-4-pull-request)
+    - [Step 5: Merge](#step-5-merge)
+  - [Getting Started](#getting-started)
+    - [Setup](#setup)
+    - [Repo Layout](#repo-layout)
+    - [Tests](#tests)
+      - [Unit tests](#unit-tests)
+        - [Write](#write)
+        - [Run](#run)
```
Comment (saragerion) [inline CONTRIBUTING.md:29]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -10,6 +10,18 @@ A suite of TypeScript utilities for AWS Lambda functions to ease adopting best p
 
 > **An AWS Developer Acceleration (DevAx) initiative by Specialist Solution Architects | aws-devax-open-source@amazon.com**
 
+## Table of contents
+
+- [AWS Lambda Powertools (TypeScript)](#aws-lambda-powertools-typescript)
+  - [Table of contents](#table-of-contents)
```
Comment (saragerion) [inline README.md:16]:
```suggestion
```

**Comment by dreamorosi** [comment]
+1 on having an explicit in-text ToC.

I use GitHub on mobile more than half of the time and it doesn't support the feature yet :(

Downside of this is that we'll have to remember to keep it updated, but hopefully these docs won't be changing all the time.

**Review by saragerion — APPROVED** [review]

**Review by dreamorosi — APPROVED** [review]

**Pull request merged**